### DISCUSSION TITLE
Variables in macros

### DISCUSSION THREAD

**Discussion by ronindev** [winder/Universal-G-Code-Sender#2387] [Help] [answered]
Hello! 

I'm new at CNC (I have hobby cnc3018) and new at UGS. Please, can you tell me, UGS macros have ability to use variables? 
I want to make macros for tool change and automatic tool length modification. 
1. Move to pre-defined X Y position using safe Z
2. Probe current tool
3. Keep in memory Z working coordinates of probing results
4. Pause and promt to change tool
5. Probe new tool 
6. Change working Z to the value from 3 step
7. Move back on XYZ using inital working coordinates

I've found similar macros in CNCjs project - https://github.com/cncjs/CNCjs-Macros/blob/master/Precision_ToolChange_Macro.txt
It uses variables to store Zpos. 
Can I do something like that in UGS?
Thanks!

**Comment by breiler** [comment, marked as answer]
No that is currently not possible
